### ISSUE TITLE
Not all dotnet-trace profiles should have Rundown and Sampling

### ISSUE THREAD

**Issue by noahfalk** [dotnet/coreclr#25268]
I assume the current implementation of the 'None' profile still does Rundown at the end, and probably the gc profiles too. Rundown can be very time consuming and large so it may not be what the user wanted. Tasks:

- [ ] Figure out which of our hard coded profiles should have rundown included
- [ ] Implement some mechanism that allows the trace client to configure rundown rather than having the runtime assume that rundown always happens.

[EDIT]: Also added sampling which is roughly same issue as rundown

**Comment by cshung** [comment]
In my opinion, making sure the sampler is proactively disabled is more important than avoiding rundown. Saving the rundown saves at most a few seconds per session, but saving the sampling profiler is going to save on average 10% overhead as I roughly measured.

**Comment by noahfalk** [comment]
> making sure the sampler is proactively disabled is more important than avoiding rundown

I agree on the priority. Sung was going to confirm that it is being disabled correctly so I don't think we have to pick one or the other - we are getting both : D
